### PR TITLE
Switching to use tensor.pack/unpack ops for data-tiling

### PR DESCRIPTION
**Pull request by hanhanW**
benchmarks: comp-stats

### CODE REVIEW

**Comment by hanhanW** [comment]
This depends on https://github.com/iree-org/iree/pull/12153 and https://reviews.llvm.org/D144151

**Comment by hanhanW** [comment]
It also depends on https://github.com/llvm/llvm-project/commit/be75cf931f3fd3f23a04d3e8f441f09a3cf08ef9

**Comment by hanhanW** [comment]
@MaheshRavishankar this is ready for review. Please take a look when you're available, thanks!

**Comment by MaheshRavishankar** [comment]
> @MaheshRavishankar this is ready for review. Please take a look when you're available, thanks!

Awesome! Thanks! looking

**Review by MaheshRavishankar — APPROVED** [review]
Awesome! Great job upstreaming things!

File: compiler/src/iree/compiler/Codegen/Common/test/tile_and_distribute_to_workgroups.mlir [hunk 1]
```diff
@@ -2275,7 +2275,7 @@ hal.executable private @elem_pack {
           linalg.yield %23, %25 : f32, f32
         } -> (tensor<384x512xf32>, tensor<384x512xf32>)
         %17 = tensor.empty() : tensor<48x512x8x1xf32>
-        %18 = iree_linalg_ext.pack {encoding = #iree_linalg_ext.encoding<MATMUL_F32F32F32_LHS>, lowering_config = #iree_codegen.lowering_config<tile_sizes = [[8, 64]]>} %16#0 inner_dims_pos = [0, 1] inner_tiles = [8, 1] into %17 : (tensor<384x512xf32> tensor<48x512x8x1xf32>) -> tensor<48x512x8x1xf32>
+        %18 = tensor.pack %16#0 inner_dims_pos = [0, 1] inner_tiles = [8, 1] into %17 {lowering_config = #iree_codegen.lowering_config<tile_sizes = [[8, 64]]>} : tensor<384x512xf32> -> tensor<48x512x8x1xf32>
```
Comment (MaheshRavishankar) [inline compiler/src/iree/compiler/Codegen/Common/test/tile_and_distribute_to_workgroups.mlir:2278]: Maybe wait for @bjacob to take a look on this one. There was something here w.r.t to why encoding was left around

Comment (hanhanW) [reply compiler/src/iree/compiler/Codegen/Common/test/tile_and_distribute_to_workgroups.mlir:2278]: The test was added by my another PR. The encoding in attribute is only used for microkernel approach. This is the test for llvm-cpu target, so I think it's fine.

**Pull request merged**